llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -1738,6 +1738,12 @@ SDValue SelectionDAGBuilder::getValueImpl(const Value *V) {
     if (const auto *NC = dyn_cast<NoCFIValue>(C))
       return getValue(NC->getGlobalValue());
 
+    if (VT == MVT::aarch64svcount) {
+      assert(C->isNullValue() && "Can only zero this target type!");
+      return DAG.getNode(ISD::BITCAST, getCurSDLoc(), VT,
```

Comment:
Hi, @paulwalker-arm .
I wanna ask whether we should insert `NodeMap[V] =` here for caching the mapping between them? Or I neglect sth? (I am still a beginner in LLVM community)
```C++
return NodeMap[V] = DAG.getNode(...
``` 
If you think it's good to have, I can open a PR to add this as my first PR!

Reply:
I agree, it looks like an omission on my part when landing the original work. If you're happy to fix then that would be great, otherwise just let me know and I'll follow up.